src/NuGet.Services.AzureSearch/Analysis/DescriptionCustomAnalyzer.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Microsoft.Azure.Search.Models;
+
+namespace NuGet.Services.AzureSearch
+{
+    /// <summary>
+    /// Support for NuGet style description analysis. This splits tokens
+    /// on non alpha-numeric characters, splits tokens on camel casing,
+    /// lower cases tokens, and then removes stopwords from tokens.
+    /// </summary>
+    public static class DescriptionAnalyzer
```

Comment:
This replaces the old search's `DescriptionAnalyzer`, but has a few differences:

1. The old analyzer split on the symbols `. - , ; : \ * # ! ~ ~ + - ( ) [ ] { }`. The new analyzer splits on any non alpha-numeric characters. For example: `test_package` used to be tokenized as `test_package`, it will now tokenize as `test` and `package`
2. The old analyzer would split numerical values differently. For example: `abc123def` used to be tokenized as `abc` and `123def`, it will now tokenize as `abc`, `123`, and `def`

These two changes are improvements to tokenization

Reply:
See [these tests](https://github.com/NuGet/NuGet.Services.Metadata/pull/487/files#diff-508433a8283902bade024b00573c7a6bR32) for tokenization examples